packages/compiler/src/formatter/print/printer.ts

```diff
@@ -1975,7 +1975,10 @@ function printItemList<T extends Node>(
  * @param options Prettier options
  * @returns Raw text in the file for the given node.
  */
-function getRawText(node: TextRange, options: TypeSpecPrettierOptions) {
+function getRawText(node: TextRange, options: TypeSpecPrettierOptions): string {
+  if ("rawText" in node) {
```

Comment:
Added this way of setting the syntax for basic nodes where they can have `rawText` instead of relying on the `pos` and `end` and the whole file making writing tsp from an AST impossible